File: javaOOP/lifeSimulator/src/Organism.java
import javax.swing.JButton;

public abstract class Organism {
    protected int age;
    protected int strength;
    protected int initiative;
    protected int cooldown;
    protected int posY;
    protected int posX;
    protected char sign;
    protected String species;
    protected World world;

    public abstract void action();
    public abstract void collision(Organism attacker);
    public abstract JButton getButton();

    public Organism(String species, char sign, int posY, int posX, int strength, int initiative, World world) {
        this.species = species;
        this.sign = sign;
        this.posY = posY;
        this.posX = posX;
        this.strength = strength;
        this.initiative = initiative;
        this.world = world;
        this.cooldown = 0;
        this.age = 0;
    }

    public char draw() {
        return sign;
    }

    public int getPosY() {
        return posY;
    }

    public int getPosX() {
        return posX;
    }

    public int getAge() {
        return age;
    }

    public int getCooldown() {
        return cooldown;
    }

    public int getInitiative() {
        return initiative;
    }

    public String getSpecies() {
        return species;
    }

    public int getStrength() {
        return strength;
    }

    public void setPosY(int posY) {
        this.posY = posY;
    }

    public void setPosX(int posX) {
        this.posX = posX;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public void setCooldown(int cooldown) {
        this.cooldown = cooldown;
    }

    public void setInitiative(int initiative) {
        this.initiative = initiative;
    }

    public void setStrength(int strength) {
        this.strength = strength;
    }

    public void advanceAge() {
        this.age++;
    }

    public void advanceCooldown() {
        if(this.cooldown > 0) this.cooldown--;
    }

    public void gainStrength() {
        this.strength += 3;
    }

    public String getInformation() {
        return this.getSpecies() + ";" + Integer.toString(this.getPosY()) + ";" + Integer.toString(this.getPosX()) +
        ";" + Integer.toString(this.getStrength()) + ";" + Integer.toString(this.getInitiative()) + ";" +
        Integer.toString(this.getAge()) + ";" + Integer.toString(this.getCooldown());
    }
    
}

File: cppFiles/Organism.cpp
#include "Organism.h"
#include "World.h"

Organism::Organism() {
	species = "";
	sign = ' ';
	posX = 0;
	posY = 0;
	strength = 0;
	cooldown = 0;
	age = 0;
	initiative = 0;
	world = nullptr;
}

Organism::Organism(string species, char sign, int posX, int posY, int strength, int initiative, World* world)
	: species{ species }, sign{ sign }, posX{ posX }, posY{ posY }, strength{ strength }, initiative{ initiative }, world{ world } {
}

char Organism::draw() {
	return sign;
}

void Organism::advanceAge() {
	this->age++;
}

void Organism::advanceCooldown() {
	if (this->cooldown != 0) this->cooldown--;
}

int Organism::getX() {
	return posX;
}

int Organism::getY() {
	return posY;
}

int Organism::getInitiative() {
	return initiative;
}

int Organism::getAge() {
	return age;
}

int Organism::getCooldown() {
	return cooldown;
}

string Organism::getSpecies() {
	return species;
}

int Organism::getStrength() {
	return strength;
}

void Organism::setCooldown(int number) {
	this->cooldown = number;
}

void Organism::setY(int number) {
	this->posY = number;
}

void Organism::setX(int number) {
	this->posX = number;
}

void Organism::setAge(int number) {
	this->age = number;
}

void Organism::setStrength(int number) {
	this->strength = number;
}

void Organism::setInitiative(int number) {
	this->initiative = number;
}

void Organism::gainStrength() {
	this->strength += 3;
}

string Organism::getInformation() {
	string temp;
	temp = this->getSpecies() + ";" + to_string(this->getX()) + ";" + to_string(this->getY()) + ";" +
		to_string(this->getStrength()) + ";" + to_string(this->getInitiative()) + ";" + to_string(this->getAge()) + ";" + 
		to_string(this->getCooldown());
	return temp;
}

bool compareOrganisms(Organism* first, Organism* second) {
	if (first->getInitiative() > second->getInitiative()) return true;
	else if (first->getInitiative() < second->getInitiative()) return false;
	else if (first->getAge() > second->getAge()) return true;
	else return false;
}




